streams/src/main/java/org/apache/kafka/streams/kstream/internals/graph/TableProcessorNode.java

```diff
@@ -65,8 +72,9 @@ public void writeToTopology(final InternalTopologyBuilder topologyBuilder) {
         }
 
         // TODO: we are enforcing this as a keyvalue store, but it should go beyond any type of stores
-        if (storeBuilder != null) {
-            topologyBuilder.addStateStore(storeBuilder, processorName);
+        if (storeNeeded) {
```

Comment:
nit: not related to this PR, but the above `TODO` can be renamed as `TODO KIP-300` to be more specific.